system-addon/content-src/components/TopSites/TopSiteFormInput.jsx

```diff
@@ -0,0 +1,45 @@
+import {FormattedMessage} from "react-intl";
+import React from "react";
+
+export class TopSiteFormInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onMount = this.onMount.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.validationError) {
+      this.input.focus();
+    }
+  }
+
+  onMount(input) {
+    this.input = input;
+  }
+
+  render() {
+    const showClearButton = this.props.value && this.props.onClear;
+
+    return (<label><FormattedMessage id={this.props.titleId} />
+      <div className={`field url${this.props.validationError ? " invalid" : ""}`}>
```

Comment:
This can't be `.url` for all inputs. Only for url type inputs, so not the title.